sql-plugin/src/main/330+/scala/org/apache/spark/sql/rapids/shims/intervalExpressions.scala

```diff
@@ -0,0 +1,497 @@
+   */
+  def checkMultiplyOverflowImp(x: BinaryOperable, y: ColumnVector,
+      r: ColumnVector): Unit = {
+    // check (y != 0) && (r/y != x)
+    checkMultiplyOverflowSimpleImp(x, y, r)
+
+    // check (x == Long.MIN_VALUE && y == -1)
+    x match {
+      case scalar: Scalar =>
+        if (getLong(scalar) == Long.MinValue) { // x == Long.MIN_VALUE
+          withResource(Scalar.fromLong(-1L)) { negOne =>
+            withResource(y.equalTo(negOne)) { negOneBool =>
+              if (negOneBool.any().getBoolean) { // y = -1L
+                throw new ArithmeticException("overflow occurs")
+              }
+            }
+          }
+        }
+      case _ =>
+        // check (x == Long.MIN_VALUE && y == -1)
+        withResource(Scalar.fromLong(Long.MinValue)) { minLong =>
+          withResource(Scalar.fromLong(-1L)) { negOneLong =>
+            withResource(x.equalTo(minLong)) { minBool =>
+              withResource(y.equalTo(negOneLong)) { oneBool =>
+                withResource(minBool.add(oneBool)) { overflow =>
+                  if (overflow.any().getBoolean) {
+                    throw new ArithmeticException("overflow occurs")
+                  }
+                }
+              }
+            }
+          }
+        }
+    }
+  }
+
+  /**
+   * Check if the long result of long * long is overflow
+   * See Math.multiplyExact:
+   * r = x * y;
+   * if (y != 0) && (r/y != x) throw exception;
+   * if(x == Long.MIN_VALUE && y == -1) throw exception
+   *
+   * Here is a little different with `Math.multiplyExact`,
+   * omits the `check x == Long.MIN_VALUE && y == -1`,
+   * Because of the `toIntWithCheck` will throw exception
+   * for the value of `Long.MIN_VALUE * -1` exceeds int limits
+   *
+   * @param leftLong  long cv or scalar
+   * @param rightLong long cv or scalar, will not be scalar if leftLong is scalar
+   * @param result    is the result of x * y
+   */
+  def checkMultiplyOverflowSimple(leftLong: BinaryOperable, rightLong: BinaryOperable,
+      result: ColumnVector): Unit = {
+    leftLong match {
+      case leftVector: ColumnVector =>
+        checkMultiplyOverflowSimpleImp(rightLong, leftVector, result)
+      case _ =>
+        checkMultiplyOverflowSimpleImp(leftLong, rightLong.asInstanceOf[ColumnVector], result)
+    }
+  }
+
+  /**
+   * Check if the long result of long * long is overflow
+   * See Math.multiplyExact:
+   * r = x * y;
+   * if (y != 0) && (r/y != x) throw exception;
+   * if(x == Long.MIN_VALUE && y == -1) throw exception
+   *
+   * Here is a little different with `Math.multiplyExact`,
+   * omits the `check x == Long.MIN_VALUE && y == -1`,
+   * Because of the `toIntWithCheck` will throw exception
+   * for the value of `Long.MIN_VALUE * -1` exceeds int limits
+   *
+   * @param x long cv or scalar
+   * @param y long cv or scalar, will not be scalar if x is scalar
+   * @param r is the result of x * y
+   */
+  def checkMultiplyOverflowSimpleImp(x: BinaryOperable, y: ColumnVector,
```

Comment:
The postfix `Imp`, personally , is not suitable here. Usually it is used in a class declaration for implementing a interface, or a concrete method declaration implementing a abstract method.

Reply:
Updated